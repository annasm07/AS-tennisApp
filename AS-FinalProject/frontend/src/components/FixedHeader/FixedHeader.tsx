import * as React from 'react';
import {StyleSheet, Image, TouchableOpacity} from 'react-native';
import {SafeAreaView} from 'react-native-safe-area-context';

function FixedHeader() {
  return (
    <SafeAreaView style={styles.header}>
      <TouchableOpacity>
        <Image
          style={styles.back}
          source={require('../../images/arrow-back.png')}
        />
      </TouchableOpacity>
      <TouchableOpacity>
        <Image
          style={styles.icon}
          source={require('../../images/ball-logo.png')}
        />
      </TouchableOpacity>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  header: {
    borderBottomColor: 'gray',
    borderBottomWidth: 1,
    marginHorizontal: 10,
    flexDirection: 'row',
  },
  back: {
    width: 30,
    height: 30,
    alignSelf: 'flex-start',
    marginBottom: -30,
    marginTop: 10,
    marginLeft: 20,
  },
  icon: {
    width: 50,
    height: 50,
    marginLeft: 70,
    marginBottom: -30,
    left: '40%',
  },
});

export default FixedHeader;
