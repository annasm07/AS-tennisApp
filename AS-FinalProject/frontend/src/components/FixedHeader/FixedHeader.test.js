import {render} from '@testing-library/react-native';
import React from 'react';
import ConfigureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import FixedHeader from './FixedHeader';

describe('When invoked a Dashboard component', () => {
  const mockStore = ConfigureStore();
  const store = mockStore({
    user: {user: {playerId: '123cde'}},
  });
  test('Should render dashboard component', () => {
    const fixedHeader = render(
      <Provider store={store}>
        <FixedHeader />
      </Provider>,
    );
    expect(fixedHeader).toMatchSnapshot();
  });
});
