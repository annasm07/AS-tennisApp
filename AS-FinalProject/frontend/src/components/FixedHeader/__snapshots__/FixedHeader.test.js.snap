// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When invoked a Dashboard component Should render dashboard component 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "borderBottomColor": "gray",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "marginHorizontal": 10,
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/arrow-back.png",
        }
      }
      style={
        Object {
          "alignSelf": "flex-start",
          "height": 30,
          "marginBottom": -30,
          "marginLeft": 20,
          "marginTop": 10,
          "width": 30,
        }
      }
    />
  </View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/ball-logo.png",
        }
      }
      style={
        Object {
          "height": 50,
          "left": "40%",
          "marginBottom": -30,
          "marginLeft": 70,
          "width": 50,
        }
      }
    />
  </View>
</RNCSafeAreaView>
`;
